### ISSUE TITLE
.singleton_methods returns an empty array after prepending

### ISSUE THREAD

**Issue by flash-gordon** [jruby/jruby#5799]
### Environment

```
$ jruby -v
jruby 9.2.6.0 (2.5.3) 2019-02-11 15ba00b Java HotSpot(TM) 64-Bit Server VM 25.201-b09 on 1.8.0_201-b09 +jit [darwin-x86_64]
$ uname -a
Darwin Nikitas-MacBook-Pro.local 18.6.0 Darwin Kernel Version 18.6.0: Thu Apr 25 23:16:27 PDT 2019; root:xnu-4903.261.4~2/RELEASE_X86_64 x86_64
```

### Expected Behavior

CRuby 2.6.3:

```ruby
obj = Object.new

def obj.foo
end

p obj.singleton_methods(false) # => [:foo]
obj.singleton_class.prepend(Module.new)
p obj.singleton_methods(false) # => [:foo]
```

### Actual Behavior

JRuby:
```ruby
obj = Object.new

def obj.foo
end

p obj.singleton_methods(false) # => [:foo]
obj.singleton_class.prepend(Module.new)
p obj.singleton_methods(false) # => []
```

I'm going to use `obj.singleton_class.instance_methods(false)`, it works fine in both cases.

**Comment by flash-gordon** [comment]
Ran on 9.2.7.0, same incorrect output.

**Comment by headius** [comment]
Confirmed on master.

**Comment by headius** [comment]
Fixed in abc3fbf8a1.

I did not find any tests for exactly this behavior in either the CRuby test suite or in ruby/spec. Could you add one to the [singleton_methods](https://github.com/jruby/jruby/blob/master/spec/ruby/core/kernel/singleton_methods_spec.rb) spec via PR? Thanks for the report!
